tchannel/tornado/connection.py

```diff
@@ -358,8 +365,15 @@ def _write(self, message):
             payload=payload
         )
         body = frame.frame_rw.write(f, BytesIO()).getvalue()
+        return self._write_queue.put(body)
 
-        return self.connection.write(body)
+    @tornado.gen.coroutine
+    def _drain_write_queue(self):
+        """Ensures ordering of our writes."""
+
+        while not self.closed:
+            body = yield self._write_queue.get()
```

Comment:
It has a timeout of None so blocks indefinitely until something is available to write. http://www.tornadoweb.org/en/stable/queues.html#tornado.queues.Queue.get